### PR TITLE
[W5.4][T10-A3]Wong Zeng Feng

### PR DESCRIPTION
**Pull request by zengfengw**

### CODE REVIEW

File: src/seedu/addressbook/data/person/DOB.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Guarantees: immutable; is valid as declared in {@link #isValidDOB(String)}
+ */
+public class DOB {
+
+    public static final String EXAMPLE = "01.01.1995";
+    public static final String MESSAGE_DOB_CONSTRAINTS =
+            "Person date of birth should be in the format DD.MM.YYYY and all numeric digits";
+    public static final String DOB_VALIDATION_REGEX = "\\d+.\\d+.\\d+";
+
+    public final String value;
```
Comment (AceCentury) [inline src/seedu/addressbook/data/person/DOB.java]: I think you might want to reconsider the use of 'final'. What if the user enters a wrong date by accident and wants to change it?

File: src/seedu/addressbook/data/person/DOB.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    public DOB(String dob, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedDOB = dob.trim();
+        if (!isValidDOB(trimmedDOB)) {
+            throw new IllegalValueException(MESSAGE_DOB_CONSTRAINTS);
+        }
+        this.value = trimmedDOB;
+    }
+
```
Comment (aaronyhsoh) [inline src/seedu/addressbook/data/person/DOB.java]: Can try throwing an exception if DOB entered is not valid eg. d/65.74.2000.
Perhaps limit the DD to 31/30/29/28 depending on month and leap year, and MM limit to 12.
Can also try to catch negative integers.

Comment (zengfengw) [reply src/seedu/addressbook/data/person/DOB.java]: Thanks for the suggestion!

File: src/seedu/addressbook/data/person/DOB.java (hunk 1 above)
Comment (zengfengw) [reply src/seedu/addressbook/data/person/DOB.java]: Okay i will take that into consideration. Thanks!

**Comment by madsonic** [comment]
#844 duplicate

**Pull request closed without merging**